tools/adventure-pack/goodies/java/src/union_find/UnionFind.java

```diff
@@ -26,7 +44,7 @@ public int find(int a) {
     return parent[a] = find(parent[a]);
   }
 
-  public boolean areConnected(int a, int b) {
-    return find(a) == find(b);
-  }
+  public boolean areConnected(int a, int b) { return find(a) == find(b); }
+
+  public int components() { return components; }
```

Comment:
I'm curious how you picked the name `components`, I've actually been thinking about what to potentially call this 😄

Reply:
The internet tends to call them components. Another source where they're called components is the Competitive Programming 4 book.

Reply:
https://stackoverflow.com/questions/28398101/union-find-or-dfs-which-one-is-better-to-find-connected-component